src/utils.ts

```diff
@@ -0,0 +1,37 @@
+export function ellipse(str: string): string {
```

Comment:
Is it a requirement that the ellipses appears in the middle of the text? This feels like it could be more robustly handled with css and `text-overflow: ellipsis;` (but IIRC, that only does ellipses at the end of the truncated string).

Reply:
This is the typical pattern in apps that work with cryptocurrencies. You want to be able to eyeball the beginning and the end of the address or hash to identify it quickly.